pilot/pkg/model/virtualservice.go

```diff
@@ -119,6 +119,8 @@ func mergeHTTPRoute(root *networking.HTTPRoute, delegate *networking.HTTPRoute)
 
 	if delegate.Name == "" {
 		delegate.Name = root.Name
```

Comment:
@rshriram Curently we donot check the name, it is possible to have duplicate names.

Below is one example from my own test with istio 1.5.1. Envoy does not report any error. One disadvantage is that the access trace maybe confusing.

```
[
    {
        "name": "http.80",
        "virtualHosts": [
            {
                "name": "*:80",
                "domains": [
                    "*",
                    "*:80"
                ],
                "routes": [
                    {
                        "name": "xuzhonghu",
                        "match": {
                            "prefix": "/",
                            "caseSensitive": true,
                            "headers": [
                                {
                                    "name": ":method",
                                    "exactMatch": "get"
                                }
                            ]
                        },
                        "route": {
                            "cluster": "outbound|80||sleep.default.svc.cluster.local",
                            "timeout": "0s",
                            "retryPolicy": {
                                "retryOn": "connect-failure,refused-stream,unavailable,cancelled,resource-exhausted,retriable-status-codes",
                                "numRetries": 2,
                                "retryHostPredicate": [
                                    {
                                        "name": "envoy.retry_host_predicates.previous_hosts"
                                    }
                                ],
                                "hostSelectionRetryMaxAttempts": "5",
                                "retriableStatusCodes": [
                                    503
                                ]
                            },
                            "maxGrpcTimeout": "0s"
                        },
                        "metadata": {
                            "filterMetadata": {
                                "istio": {
                                    "config": "/apis/networking.istio.io/v1alpha3/namespaces/default/virtual-service/httpbin"
                                }
                            }
                        },
                        "decorator": {
                            "operation": "sleep.default.svc.cluster.local:80/*"
                        }
                    },
                    {
                        "name": "xuzhonghu",
                        "match": {
                            "prefix": "/"
                        },
                        "route": {
                            "cluster": "outbound|8000||httpbin.default.svc.cluster.local",
                            "timeout": "0s",
                            "retryPolicy": {
                                "retryOn": "connect-failure,refused-stream,unavailable,cancelled,resource-exhausted,retriable-status-codes",
                                "numRetries": 2,
                                "retryHostPredicate": [
                                    {
                                        "name": "envoy.retry_host_predicates.previous_hosts"
                                    }
                                ],
                                "hostSelectionRetryMaxAttempts": "5",
                                "retriableStatusCodes": [
                                    503
                                ]
                            },
                            "maxGrpcTimeout": "0s"
                        },
                        "metadata": {
                            "filterMetadata": {
                                "istio": {
                                    "config": "/apis/networking.istio.io/v1alpha3/namespaces/default/virtual-service/httpbin"
                                }
                            }
                        },
                        "decorator": {
                            "operation": "httpbin.default.svc.cluster.local:8000/*"
                        }
                    }
                ]
            }
        ],
        "validateClusters": false
    }
]

```